src/coreclr/src/debug/createdump/crashinfo.cpp

```diff
@@ -652,29 +633,34 @@ CrashInfo::UnwindAllThreads(IXCLRDataProcess* pClrDataProcess)
 // Replace an existing module mapping with one with a different name.
 //
 void
-CrashInfo::ReplaceModuleMapping(CLRDATA_ADDRESS baseAddress, const char* pszName)
+CrashInfo::ReplaceModuleMapping(CLRDATA_ADDRESS baseAddress, ULONG64 size, const std::string& pszName)
 {
-    // Add or change the module mapping for this PE image. The managed assembly images are
-    // already in the module mappings list but in .NET 2.0 they have the name "/dev/zero".
-    MemoryRegion region(PF_R | PF_W | PF_X, (ULONG_PTR)baseAddress, (ULONG_PTR)(baseAddress + PAGE_SIZE), 0, pszName);
-    const auto& found = m_moduleMappings.find(region);
+    uint64_t start = (uint64_t)baseAddress;
+    uint64_t end = ((baseAddress + size) + (PAGE_SIZE - 1)) & PAGE_MASK;
+
+    // Add or change the module mapping for this PE image. The managed assembly images may already
+    // be in the module mappings list but they may not have the full assembly name (like in .NET 2.0
+    // they have the name "/dev/zero"). On MacOS, the managed assembly modules have not been added.
+    MemoryRegion search(0, start, start + PAGE_SIZE);
```

Comment:
We used to search with all flags, now we search with no flags. Does this have any impact on a remap? I don't know what cases need this remap.

Reply:
The "find" only searches on the start/end addresses (it uses the `operator<` in the MemoryRegion) so the flags and the rest don't matter.  Originally, this was to fix up the memory mapped assembly names in the 1.1 runtime (they were all /dev/zero) and became moot in 2.x because the mapped memory had the proper assembly path. Now it is relevant again on MacOS to add the assemblies to the native module list like our tools expect. I left the "replace" name condition (added a name compare) just to protect from any future runtime changes on Linux.  

FYI, on Linux we use the /proc/<pid>/maps to populate the moduleMappings list which contains both native modules and managed assemblies because the later uses mapped memory.  On MacOS, we the dylinker cache which just has the native modules.  Technically on Linux we could be using the "DSO" module list which is what native debuggers like lldb/gdb (which is why createdump enumerates and add the memory).

Reply:
Yeah, there's cases I've gone through the DSO list to resolve some symbols before under lldb, and that was definitely useful.